src/mux/comment.ts

```diff
@@ -0,0 +1,354 @@
+          },
+        };
+      }
+    }
+    throw new Error('We are not able to save comment');
+  },
+);
+
+// Edit comment in article
+Mux.put<IComment>(
+  '/v1/comment/:commentUuid',
+  CommentValidator,
+  async (requestData: IRequestData, req?: express.Request): Promise<IResponseRecord<IComment>> => {
+    const { commentUuid } = requestData.params;
+    if (req) {
+      // Get userId from header
+      const userId = req.header('X-Denote-User-Identity');
+      // Find is user existing
+      const user = await ModelUser.findOne({ userId });
+      const comment = await ModelComment.findOne({ uuid: commentUuid });
+      if (comment && user) {
+        if (comment.author === user._id) {
+          const savedComment = await ModelComment.findOneAndUpdate(
+            { uuid: commentUuid, author: user._id },
+            { ...requestData.body },
+          );
+          if (savedComment) {
+            const {
+              uuid,
+              author,
+              articleId,
+              reply,
+              votedUser,
+              content,
+              created,
+              updated,
+              hidden,
+              vote,
+            } = savedComment.toObject({
+              transform: (_doc: any, ret: any) => {
+                const keys = Object.keys(ret);
+                for (let i = 0; i < keys.length; i += 1) {
+                  const key = keys[i];
+                  if (key.indexOf('_') === 0) {
+                    // eslint-disable-next-line no-param-reassign
+                    delete ret[key];
+                  }
+                }
+              },
+            });
+            return {
+              success: true,
+              result: {
+                uuid,
+                articleId,
+                author,
+                reply,
+                votedUser,
+                content,
+                created,
+                updated,
+                hidden,
+                vote,
+              },
+            };
+          }
+        }
+      }
+    }
+    throw new Error('We are not able to save comment');
+  },
+);
+
+// Delete a comment
+Mux.delete<IComment>(
+  '/v1/comment/:commentUuid',
+  UuidValidator,
+  async (requestData: IRequestData, req?: express.Request): Promise<IResponseRecord<IComment>> => {
+    const { commentUuid } = requestData.params;
+    if (req) {
+      // Get userId from header
+      const userId = req.header('X-Denote-User-Identity');
+      // Find is user existing
+      const user = await ModelUser.findOne({ userId });
+      if (user) {
+        const deletedComment = await ModelComment.findOneAndDelete({ uuid: commentUuid, author: user._id });
+        if (deletedComment) {
+          const {
+            uuid,
+            author,
+            articleId,
+            reply,
+            votedUser,
+            content,
+            created,
+            updated,
+            hidden,
+            vote,
+          } = deletedComment.toObject({
```

Comment:
Why do we need this?

Reply:
I think we should strictly match the IComment Interface, isn't it?

Reply:
This code is unnecessary, might be it here since I was try to test `transformation` then you guys misunderstand then copy it here again.

```ts
transform: (_doc: any, ret: any) => {
              const keys = Object.keys(ret);
              for (let i = 0; i < keys.length; i += 1) {
                const key = keys[i];
                if (key.indexOf('_') === 0) {
                  // eslint-disable-next-line no-param-reassign
                  delete ret[key];
                }
              }
```

```ts
const  {uuid}  = comment.toObject();
````

Reply:
Yes, first time read this code, I don't know why it was here, will fix it later this night.